packages/autocomplete-shared/src/decycle.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Decycles objects with circular references.
+ * This is used to print cyclic structures in development environment only.
+ */
+export function decycle(obj: any, seen: any[] = []) {
+  if (!__DEV__ || !obj || typeof obj !== 'object') {
+    return obj;
+  }
+
+  if (seen.includes(obj)) {
```

Comment:
why is seen not a WeakMap like in cycle.js?

Reply:
This isn't necessary in our case because we don't need to keep track of any `path` as they do in cycle.js. However, a `Set` might be a better fit for easier and faster access! wdyt?

Reply:
Addressed in 1c63db79.